huaweicloud/services/evs/resource_huaweicloud_evs_volume.go

```diff
@@ -402,16 +446,23 @@ func resourceEvsVolumeDelete(ctx context.Context, d *schema.ResourceData, meta i
 		}
 	}
 
-	// The snapshots associated with the disk are deleted together with the EVS disk if cascade value is true
-	deleteOpts := cloudvolumes.DeleteOpts{
-		Cascade: d.Get("cascade").(bool),
-	}
-	// It's possible that this volume was used as a boot device and is currently
-	// in a "deleting" state from when the instance was terminated.
-	// If this is true, just move on. It'll eventually delete.
-	if v.Status != "deleting" {
-		if err := cloudvolumes.Delete(evsV2Client, d.Id(), deleteOpts).ExtractErr(); err != nil {
-			return common.CheckDeletedDiag(d, err, "EVS volume")
+	if d.Get("charging_mode").(string) == "prePaid" {
+		err = common.UnsubscribePrePaidResource(d, config, []string{d.Id()})
+		if err != nil {
+			return fmtp.DiagErrorf("error unsubscribing HuaweiCloud EVS volume : %s", err)
+		}
+	} else {
```

Comment:
should we try to delete the disk after unsubcribe it?

Reply:
Prepaid disks will be completely removed after the order is completed.